src/helpers/covalent/interpretTransaction.ts

```diff
@@ -0,0 +1,151 @@
+  Object.values(contract.addresses)
+    .map(address => address.toLowerCase())
+    .includes(address.toLowerCase());
+
+export const interpretTransaction = (transactions: CovalentTransaction[], address: string) => {
+  const results: Transaction[] = [];
+
+  for (const transaction of transactions) {
+    if (!transaction.log_events || transaction.log_events.length === 0) continue;
+
+    assert(transaction.log_events, "Transactions w/o logs are ignored");
+    const [first, second] = transaction.log_events;
+
+    if (isContract(BOND_DEPOSITORY_CONTRACT, transaction.to_address)) {
+      if (first.decoded.params[1].value.toLowerCase() === address.toLowerCase())
+        results.push({
+          token: OHM_TOKEN,
+          transaction,
+          type: "bond",
+          details: "Bond Claimed",
+          value: new DecimalBigNumber(BigNumber.from(first.decoded.params[2].value), first.sender_contract_decimals),
+        });
+      else
+        results.push({
+          token: OHM_TOKEN,
+          transaction,
+          type: "bond",
+          details: "Bond Purchased",
+          value: new DecimalBigNumber(BigNumber.from(second.decoded.params[2].value), second.sender_contract_decimals),
+        });
+    }
+
+    if (isContract(STAKING_CONTRACT, transaction.to_address)) {
+      if (isContract(STAKING_CONTRACT, first.decoded.params[0].value) && isContract(OHM_TOKEN, first.sender_address))
+        results.push({
+          token: OHM_TOKEN,
+          transaction,
+          type: "staking",
+          details: "Unstake",
+          value: new DecimalBigNumber(BigNumber.from(first.decoded.params[2].value), first.sender_contract_decimals),
+        });
+      else
+        results.push({
+          token: OHM_TOKEN,
+          transaction,
+          type: "staking",
+          details: "Stake",
+          value: new DecimalBigNumber(BigNumber.from(first.decoded.params[2].value), first.sender_contract_decimals),
+        });
+    }
+
+    if (isContract(ZAP_CONTRACT, transaction.to_address))
+      results.push({
+        token: OHM_TOKEN,
+        transaction,
+        type: "zap",
+        details: `Zap to ${second?.sender_contract_ticker_symbol}`,
+        value: new DecimalBigNumber(BigNumber.from(second.decoded.params[2].value), second.sender_contract_decimals),
+      });
+
+    if (isContract(MIGRATOR_CONTRACT, transaction.to_address))
+      results.push({
+        token: OHM_TOKEN,
+        transaction,
+        type: "migration",
+        details: "Migration",
+        value: new DecimalBigNumber(BigNumber.from(first.decoded.params[2].value), first.sender_contract_decimals),
+      });
+
+    if (isContract(PT_PRIZE_POOL_CONTRACT, transaction.to_address))
+      results.push({
+        token: OHM_TOKEN,
+        transaction,
+        type: "33together",
+        details: "33Together Claim",
+        value: new DecimalBigNumber(BigNumber.from(second.decoded.params[2].value), second.sender_contract_decimals),
+      });
```

Comment:
this contract has been in active for 3-4 months. What do you think about removing it? maybe in a separate pr

Reply:
I guess there's no harm in keeping it.